src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -239,6 +241,99 @@ public void FormulaCantRedefine()
                engine.SetFormula("A", "3", OnUpdate));
         }
 
+        [Fact]
+        public void DefFunc()

```

Comment:
Great test cases! Consider also adding tests for 
Function Bodies with errors,
Definitions with the same name,
invocations with incorrect parameter count/types

Reply:
Done